File: python/shared/dtos/preset_member_dto.py
from . import BaseDTO
from .member_dto import MemberDetailDTO
from .preset_member_position_dto import PresetMemberPositionDetailDTO
from .tier_dto import TierDTO


class PresetMemberDTO(BaseDTO):
    preset_member_id: int
    preset_id: int
    member_id: int
    tier_id: int | None
    is_leader: bool


class PresetMemberDetailDTO(PresetMemberDTO):
    member: MemberDetailDTO
    tier: TierDTO | None
    preset_member_positions: list[PresetMemberPositionDetailDTO]


class AddPresetMemberDTO(BaseDTO):
    member_id: int
    tier_id: int | None
    is_leader: bool


class UpdatePresetMemberDTO(BaseDTO):
    tier_id: int | None = None
    is_leader: bool | None = None

File: web/src/dtos/presetMemberDto.ts
import type { MemberDetailDTO } from "./memberDto";
import type { PresetMemberPositionDetailDTO } from "./presetMemberPositionDto";
import type { TierDTO } from "./tierDto";

export interface PresetMemberDTO {
  presetMemberId: number;
  presetId: number;
  memberId: number;
  tierId: number | null;
  isLeader: boolean;
}

export interface PresetMemberDetailDTO extends PresetMemberDTO {
  member: MemberDetailDTO;
  tier: TierDTO | null;
  presetMemberPositions: PresetMemberPositionDetailDTO[];
}

export interface AddPresetMemberDTO {
  memberId: number;
  tierId: number | null;
  isLeader: boolean;
}

export interface UpdatePresetMemberDTO {
  tierId?: number | null;
  isLeader?: boolean;
}
